### ISSUE TITLE
Initial Initialization Issue

### ISSUE THREAD

**Issue by bilbobaggins30** [mbdavid/LiteDB#1114]
Firstly, I want to say this project is a gem! It works great in C#, and I love how cleverly it works.

However, whenever a database is created for the first time, you have to call the insert method on the collection in order to actually create the .db file.

My situation that this came up, is I have to design around the fact that a user may move the .db files to another location, some .db files may come up missing, and the user may start the program for the 1st time, just to close out of it after it starts. 

So I spend time checking that all the files required by the program are there. However, I found that I have to insert "dummy" data into the database to write the file. Is there a way this can be fixed? I would like to correct this issue on my end, if you can point me to the source code that I could modify to correct this behavior. 

Thanks in advance! Also, ALL OF YOU GUYS ROCK! This solution beats out SQLite and other solutions for storing data that I have spent too much time fighting to get to work! This is about as clean as it gets from the library user's aspect, and I appreciate the hard work!

**Comment by mbdavid** [comment]
Hi @bilbobaggins30, thanks for your comments! Its nice read this!

This occurs because you are using `LiteDatabase` class, and this class contains a `Lazy<LiteEngine>` instance. Only after any use database will be open (or created).

But, you can:

1) Use `LiteEngine` to open database (or create). This will create datafile at this ctor moment.

or

2) Call `LiteEngine.CreateDatabase(Stream stream, string password = null, long initialSize = 0)` static method.

or

3) Get/Set a simple `UserVersion` property inside LiteEngine, when using LiteDatabase class: `db.Engine.UserVersion = 1;`

**Comment by bilbobaggins30** [comment]
@mbdavid Hey I just wanted to ping you again. Setting the UserVersion cleaned things up NICELY for me! Up to you if you want to leave this issue open, or close it, and add it to the Docs for others to have this information. 

Thanks again!

**Comment by lbnascimento** [comment]
Hi! With the objective of organizing our issues, we are closing old unsolved issues. Please check the latest version of LiteDB and open a new issue if your problem/question/suggestion still applies. Thanks!
